kubexecutor/src/main/kotlin/pl/edu/prz/kod/executor/JavaExecutor.kt

```diff
@@ -0,0 +1,42 @@
+package pl.edu.prz.kod.executor
+
+import pl.edu.prz.kod.domain.ExecutionResult
+import pl.edu.prz.kod.domain.Language
+import pl.edu.prz.kod.error.CompilationFailedError
+import java.io.File
+import java.util.regex.Pattern
+
+class JavaExecutor: AbstractExecutor(Language.JAVA) {
+    private val srcDirPath = "/app/executor/src"
+    private val buildDirPath = "/app/executor/build"
+    private val runDirPath = "/app/executor/run"
+
+    override fun execute(code: String): ExecutionResult {
+        val srcDir = File(srcDirPath)
+        srcDir.mkdirs()
+        val runDir = File(runDirPath)
+        runDir.mkdirs()
+
+        val classNameMatcher = Pattern.compile("class\\s+([^\\s{]+)[\\s{]").matcher(code)
+        var javaClassName = "Main"
+        if (classNameMatcher.find()) {
+            javaClassName = classNameMatcher.group(1)
+        }
+
+        val javaFile = srcDir.resolve("$javaClassName.java")
+        javaFile.createNewFile()
+        javaFile.writeText(code)
+
+        val compilationProcess = runSystemCommand("javac -d $buildDirPath ${javaFile.name}", srcDir)
+        if (compilationProcess.exitValue() != 0) {
+            throw CompilationFailedError(compilationProcess.errorStream.bufferedReader().readText())
+        }
```

Comment:
How about extracting this to a function such as `compileCode`? It'd look prettier imo though it's up to You.

Reply:
Honestly, I would prefer to keep it this way, because:
1. Function would be just a few lines long, but would require 3 arguments.
2. It would then be good idea to split the remaining code into similar subfunctions, but again those would be quite short and would require a few arguments, which imo might reduce readibility.